File: SparkStreaming/src/main/java/com/projects/bigdata/spark_streaming/utility/StreamingAppFactory.java
package com.projects.bigdata.spark_streaming.utility;

import com.projects.bigdata.spark_streaming.AbstractStreaming;
import com.projects.bigdata.spark_streaming.StreamingWordsCount;
import com.projects.bigdata.spark_streaming.TradesAnalytics;
import com.projects.bigdata.utility.StreamingLineType;

public final class StreamingAppFactory {

	public static AbstractStreaming getStreamingApp(final StreamingLineType slt, final StreamingAppParameters sap) {
		switch(slt) {
		case PHRASE: return new StreamingWordsCount(sap);
		case TRADE: return new TradesAnalytics(sap);
		default: throw new IllegalArgumentException("Unknown applicaton type: "+slt);
		}
	}
}

File: SparkStreaming/src/main/kotlin/com/projects/bigdata/spark_streaming/utility/StreamingAppFactory.kt
package com.projects.bigdata.spark_streaming.utility

import com.projects.bigdata.spark_streaming.*
import com.projects.bigdata.utility.StreamingLineType

object StreamingAppFactory {

    fun getStreamingApp(slt: StreamingLineType, sap: StreamingAppParameters): AbstractStreaming = when (slt) {
            StreamingLineType.PHRASE -> StreamingWordsCount(sap)
            StreamingLineType.TRADE -> TradesAnalytics(sap)
            else -> throw IllegalArgumentException("Unknown application type: $slt")
    }
}
